### ISSUE TITLE
Convert shape's stroke into a filled shape

### ISSUE THREAD

**Issue by Keavon** [GraphiteEditor/Graphite#602]
There should be a button that can be used to convert the selected shape layer(s) into layers which have a stroke converted into a filled shape (probably with negative space inside). If the starting layer has both a fill and a shape, this should become a folder containing one layer for a shape with just the original fill and another layer with just the shape for the original stroke.

This may require an algorithm for converting a stroke into a filled shape if our libraries don't provide such a capability.

**Comment by 0HyperCube** [comment]
> There should be a button that can be used to convert the selected shape layer(s) into layers which have a stroke converted into a filled shape

@Keavon Can you not just toggle the fill colour and (if you want) disable the stroke colour?

> If the starting layer has both a fill and a shape, this should become a folder containing one layer for a shape with just the original fill and another layer with just the shape for the original stroke.

This seems a bit useless - can't you just duplicate the layer and disable the stroke on one and disable the fill on the other and then group them?

**Comment by Keavon** [comment]
The purpose is to edit the stroke as a bezier curve. Frequently when doing vector art, you will start with a stroke then turn it into its own filled shape layer and edit it. If that original shape had both a stroke and a fill, you'll want them both separated (since they are both their own separate shape layers now).

**Comment by Keavon** [comment]
Blocked on full shape-based curve expansion including caps and joins in our bezier-rs library.

**Comment by 0HyperCube** [comment]
Duplicate of #1092
